extensions/applicationinsights-osplugin-js/src/OsPlugin.ts

```diff
@@ -0,0 +1,257 @@
+                            if (!response.rejected) {
+                                _platformVersionResponse = response.value;
+                                _retrieveFullVersion = true;
+                                if (_platformVersionResponse.platformVersion && _platformVersionResponse.platform) {
+                                    _os = _platformVersionResponse.platform;
+                                    _osVer = parseInt(_platformVersionResponse.platformVersion);
+                                    if (_os === "Windows"){
+                                        if (_osVer == 0){
+                                            _osVer = 8;
+                                        } else if (_osVer < 13){
+                                            _osVer = 10;
+                                        } else{
+                                            _osVer = 11;
+                                        }
+                                    }
+                                    utlSetSessionStorage(_core.logger, "ai_osplugin", JSON.stringify({platform: _os, platformVersion: _osVer}));
+                                }
+                            } else {
+                                _throwInternal(_core.logger,
+                                    eLoggingSeverity.CRITICAL,
+                                    _eInternalMessageId.PluginException,
+                                    "Could not retrieve operating system: " + response.reason);
+                            }
+                            _completeOsRetrieve();
+                        });
+                    }
+                }
+            }
+
+            function updateTeleItemWithOs(event: ITelemetryItem) {
+                if (_retrieveFullVersion){
+                    let extOS = getSetValue(getSetValue(event, strExt), Extensions.OSExt);
+                    if (_mergeOsNameVersion){
+                        setValue(extOS, "osVer", _os + _osVer, isString);
+                    } else {
+                        setValue(extOS, "osVer", _osVer);
+                        setValue(extOS, "os", _os, isString);
+                    }
+                }
+            }
+        
+            /**
+            * Complete retrieving operating system info process, cleanup and release the event queue
+            */
+            function _completeOsRetrieve() {
+                if (_getOSTimeout) {
+                    clearTimeout(_getOSTimeout);
+                }
+                _getOSInProgress = false;
+                _releaseEventQueue();
+            }
+        
+            /**
+            * Release internal event queue
+            */
+            function _releaseEventQueue() {
+                arrForEach(_eventQueue, (evt) => {
+                    updateTeleItemWithOs(evt.item);
+                    evt.ctx.processNext(evt.item);
```

Comment:
This should really be `_self.processNext(event, itemCtx)`, while "technically" this will call the `itemCtx.processNext` as here. It's actually possible that the `ctx` is undefined or null (because of backward compatability), this is because depending on the extension that is "loaded" before this one it "may" pre-date the introduction of the `processNext` and instead just be the old "chained" `processTelemetry` calls.